### PR TITLE
test: update dep versions to use cloud-lts-bom

### PR DESCRIPTION
**Pull request by frankyn**
Validate build passes when using cloud-lts-bom dependencies. DO NOT MERGE. Not meant to be merged, only testing.

### CODE REVIEW

**Review by elharo — CHANGES_REQUESTED** [review]
to prevent merges mark the PR as draft

File: google-cloud-storage/pom.xml [hunk 1]
```diff
@@ -98,21 +104,25 @@
     <dependency>
       <groupId>io.grpc</groupId>
       <artifactId>grpc-api</artifactId>
+      <version>1.36.1</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>io.grpc</groupId>
       <artifactId>grpc-netty-shaded</artifactId>
+      <version>1.36.1</version>
```
Comment (elharo) [inline google-cloud-storage/pom.xml:113]: gRPC is now at 1.36.2 for LTS

Comment (frankyn) [reply google-cloud-storage/pom.xml:113]: Running `mvn clean compile` locally and **not** turning off enforcer:
grpc-context is managed at 1.36.0

```
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for com.google.http-client:google-http-client-gson:1.39.0 paths to dependency are:
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.api-client:google-api-client:1.31.3-sp.1
    +-com.google.http-client:google-http-client-gson:1.39.0
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core:1.94.3
    +-com.google.http-client:google-http-client-gson:1.39.0
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.auth:google-auth-library-oauth2-http:0.25.2-sp.1
    +-com.google.http-client:google-http-client-gson:1.39.0 (managed) <-- com.google.http-client:google-http-client-gson:1.39.1
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core:1.94.3
    +-com.google.http-client:google-http-client-gson:1.39.0 (managed) <-- com.google.http-client:google-http-client-gson:1.39.1
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core-http:1.94.3
    +-com.google.api:gax-httpjson:0.79.0
      +-com.google.http-client:google-http-client-gson:1.39.0
, 
Require upper bound dependencies error for com.google.auth:google-auth-library-credentials:0.24.1 paths to dependency are:
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.auth:google-auth-library-credentials:0.24.1
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core:1.94.3
    +-com.google.auth:google-auth-library-credentials:0.24.1 (managed) <-- com.google.auth:google-auth-library-credentials:0.24.0
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core-http:1.94.3
    +-com.google.auth:google-auth-library-credentials:0.24.1
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.auth:google-auth-library-oauth2-http:0.25.2-sp.1
    +-com.google.auth:google-auth-library-credentials:0.24.1 (managed) <-- com.google.auth:google-auth-library-credentials:0.25.2-sp.1
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-io.grpc:grpc-auth:1.36.2
    +-com.google.auth:google-auth-library-credentials:0.24.1 (managed) <-- com.google.auth:google-auth-library-credentials:0.22.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core:1.94.3
    +-com.google.auth:google-auth-library-credentials:0.24.1 (managed) <-- com.google.auth:google-auth-library-credentials:0.22.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.cloud:google-cloud-core-http:1.94.3
    +-com.google.api:gax-httpjson:0.79.0
      +-com.google.auth:google-auth-library-credentials:0.24.1 (managed) <-- com.google.auth:google-auth-library-credentials:0.24.0
, 
Require upper bound dependencies error for io.grpc:grpc-context:1.36.0 paths to dependency are:
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-io.opencensus:opencensus-api:0.28.0
    +-io.grpc:grpc-context:1.36.0 (managed) <-- io.grpc:grpc-context:1.27.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-io.grpc:grpc-api:1.36.2
    +-io.grpc:grpc-context:1.36.0 (managed) <-- io.grpc:grpc-context:1.36.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-io.grpc:grpc-auth:1.36.2
    +-io.grpc:grpc-context:1.36.0 (managed) <-- io.grpc:grpc-context:1.36.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.api.grpc:grpc-google-cloud-kms-v1:0.88.0
    +-io.grpc:grpc-context:1.36.0 (managed) <-- io.grpc:grpc-context:1.36.2
and
+-com.google.cloud:google-cloud-storage:1.113.14-sp.2-SNAPSHOT
  +-com.google.api.grpc:grpc-google-iam-v1:1.0.10
    +-io.grpc:grpc-context:1.36.0 (managed) <-- io.grpc:grpc-context:1.36.2

```

**Comment by frankyn** [comment]
Met with @elharo and @BenWhitehead about failures:

1. maven enforcer failing in PR#849 is acceptable for lts-bom use-case.
1. java 7 failure with java.util.function.Function is occurring in PR#849 because lts-bom uses jre version of guava which is Java 8+ and a req for the bom.

Resolved issues through discussion and closing this PR, thank you both!

**Pull request closed without merging**